clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -29,15 +33,25 @@
      * @param topic The topic this record is received from
      * @param partition The partition of the topic this record is received from
      * @param offset The offset of this record in the corresponding Kafka partition
+     * @param timestamp The timestamp of the record.
+     * @param timestampType The timestamp type
      * @param key The key of the record, if one exists (null is allowed)
      * @param value The record contents
      */
-    public ConsumerRecord(String topic, int partition, long offset, K key, V value) {
+    public ConsumerRecord(String topic,
```

Comment:
yes.  It's a minor change to update anywhere these constructors are used in unit tests, but any test that instantiates a `ConsumerRecord` isn't going to compile as soon as a user upgrades past `0.9.0.0`.  We've used the `@VisibleForTesting` annotation in the past to denote certain constructors that are only there for unit-testing.